sagemaker-spark/pyspark_mnist/xgboost_mnist.ipynb

```diff
@@ -0,0 +1,273 @@
+    "\n",
+    "The following cell runs a training job and creates an endpoint to host the resulting model, so this cell can take up to twenty minutes to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "from sagemaker_pyspark import IAMRole, S3DataPath\n",
+    "from sagemaker_pyspark.algorithms import XGBoostSageMakerEstimator\n",
+    "\n",
+    "xgboost_estimator = XGBoostSageMakerEstimator(\n",
+    "    sagemakerRole=IAMRole(role),\n",
+    "    trainingInstanceType='ml.c4.2xlarge',\n",
+    "    trainingInstanceCount=1,\n",
+    "    endpointInstanceType='ml.c4.xlarge',\n",
+    "    endpointInitialInstanceCount=1)\n",
+    "\n",
+    "xgboost_estimator.setEta(0.2)\n",
+    "xgboost_estimator.setGamma(4)\n",
+    "xgboost_estimator.setMinChildWeight(6)\n",
+    "xgboost_estimator.setSilent(0)\n",
+    "xgboost_estimator.setObjective(\"multi:softmax\")\n",
+    "xgboost_estimator.setNumClasses(10)\n",
+    "xgboost_estimator.setNumRound(10)\n",
+    "\n",
+    "# train\n",
+    "model = xgboost_estimator.fit(trainingData)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Inference\n",
+    "Now we transform our DataFrame.\n",
+    "To do this, we serialize each row's \"features\" Vector of Doubles into LibSVM format for inference against the Amazon SageMaker Endpoint. We deserialize the CSV responses from the XGBoost model back into our DataFrame:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "transformedData = model.transform(trainingData)\n",
+    "\n",
+    "transformedData.show(1000)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pyspark.sql.types import DoubleType\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "# helper function to display a digit\n",
+    "def show_digit(img, caption='', xlabel='', subplot=None):\n",
+    "    if subplot==None:\n",
+    "        _,(subplot)=plt.subplots(1,1)\n",
+    "    imgr=img.reshape((28,28))\n",
+    "    subplot.axes.get_xaxis().set_ticks([])\n",
+    "    subplot.axes.get_yaxis().set_ticks([])\n",
+    "    plt.title(caption)\n",
+    "    plt.xlabel(xlabel)\n",
+    "    subplot.imshow(imgr, cmap='gray')\n",
+    "\n",
+    "images = np.array(transformedData.select(\"features\").cache().take(250))\n",
+    "clusters = transformedData.select(\"prediction\").cache().take(250)\n",
+    "\n",
+    "for cluster in range(10):\n",
+    "    print('\\n\\n\\nCluster {}:'.format(int(cluster)))\n",
+    "    digits = [ img for l, img in zip(clusters, images) if int(l.prediction) == cluster ]\n",
+    "    height=((len(digits)-1)//5)+1\n",
```

Comment:
there's a [pep8 recommendation](https://www.python.org/dev/peps/pep-0008/#other-recommendations) to put whitespace around operators with the lowest priority, i.e.
```python
((len(digits) - 1)//5) + 1
```
YMMV.